core/src/main/kotlin/io/zeebe/bpmnspec/builder/SpecBuilder.kt

```diff
@@ -0,0 +1,141 @@
+    }
+
+    fun cancelInstance(workflowInstance: String? = null) {
+        actions.add(CancelInstanceAction(workflowInstance));
+    }
+
+    fun completeTask(jobType : String, variables: Map<String, Any> = Collections.emptyMap()) {
+        actions.add(CompleteTaskAction(jobType, serializeToJSON(variables)))
+    }
+
+    fun createInstance(bpmnProcessId: String, variables: Map<String, Any> = Collections.emptyMap(), workflowInstanceAlias: String? = null ) {
+        actions.add(CreateInstanceAction(bpmnProcessId, serializeToJSON(variables), workflowInstanceAlias))
+    }
+
+    fun publishMessage(messageName: String, correlationKey: String, variables: Map<String, Object> = Collections.emptyMap()) {
+        actions.add(PublishMessageAction(messageName, correlationKey, serializeToJSON(variables)))
+    }
+
+    fun throwError(jobType: String, errorCode: String, errorMessage: String = "" ) {
+        actions.add(ThrowErrorAction(jobType, errorCode, errorMessage));
+    }
+
+    fun build(): List<Action> {
+        return actions;
+    }
+}
+
+@TestSpecMarker
+class VerificationsBuilder {
+    private val verifications = ArrayList<Verification>()
+
+    fun elementInstanceState(selector: ElementSelector, state: ElementInstanceState, workflowInstance: String? = null) {
+        verifications.add(ElementInstanceStateVerification(state, selector.elementId, selector.elementName, workflowInstance))
+    }
+
+    fun workflowInstanceState(state: WorkflowInstanceState, workflowInstance: String? = null) {
+        verifications.add(WorkflowInstanceStateVerification(state, workflowInstance))
+    }
+
+    fun workflowInstanceVariable(variableName: String, variableValue: String, selector: ElementSelector, workflowInstance: String? = null) {
+        verifications.add(WorkflowInstanceVariableVerification(variableName, variableValue, workflowInstance, selector.elementId, selector.elementName))
+    }
+
+    fun workflowInstanceVariables(variables : Map<String, String>, selector: ElementSelector, workflowInstance: String? = null) {
+        variables.forEach { (key, value) -> workflowInstanceVariable(key, value, selector, workflowInstance) };
+    }
+
+    fun noWorkflowInstanceVariable(variableName: String, selector: ElementSelector, workflowInstance: String? = null) {
+        verifications.add(NoWorkflowInstanceVariableVerification(variableName, workflowInstance, selector.elementId, selector.elementName))
+    }
+
+    fun incidentState(state: IncidentState, errorType: String, errorMessage: String?, selector: ElementSelector, workflowInstance: String?) {
+        verifications.add(IncidentStateVerification(state, errorType, errorMessage, selector.elementId, selector.elementName, workflowInstance))
+    }
+
+    fun build(): List<Verification> {
+        return verifications
+    }
+}
+
+fun serializeToJSON( input: Any) : String {
+    return ObjectMapper().writeValueAsString(input);
+}
```

Comment:
We can set the visibility to `private` and initialize the `ObjectMapper` only once.